doc/api/esm.md

```diff
@@ -1160,29 +1160,30 @@ condition list **must** be passed through to the `defaultResolve` function.
 ```js
 /**
  * @param {string} specifier
- * @param {object} context
- * @param {string} context.parentURL
- * @param {string[]} context.conditions
- * @param {function} defaultResolve
- * @returns {object} response
- * @returns {string} response.url
+ * @param {{
+ *   parentURL:!(string|undefined),
+ *   conditions:Array<string>,
+ * }} context
+ * @param {Function} defaultResolve
+ * @returns {{url:string}}
  */
-export async function resolve(specifier, context, defaultResolve) {
+export function resolve(specifier, context, defaultResolve) {
   const { parentURL = null } = context;
-  if (someCondition) {
+  if (Math.random()) { // Some condition.
     // For some or all specifiers, do some custom logic for resolving.
     // Always return an object of the form {url: <string>}
     return {
-      url: (parentURL) ?
-        new URL(specifier, parentURL).href : new URL(specifier).href
+      url: parentURL ?
+        new URL(specifier, parentURL).href :
+        new URL(specifier).href,
     };
   }
-  if (anotherCondition) {
-    // When calling the defaultResolve, the arguments can be modified. In this
+  if (Math.random()) { // Another condition.
+    // When calling `defaultResolve`, the arguments can be modified. In this
     // case it's adding another value for matching conditional exports.
     return defaultResolve(specifier, {
       ...context,
-      conditions: [...context.conditions, 'another-condition'],
+      conditions: context.conditions.concat(['another-condition']),
```

Comment:
Why is this an improvement?

Reply:
I was seeing problems with the spread operator in Closure Compiler, but I will double-check that.

Reply:
```console
archetypes/hooks--resolve.sample.mjs:50: WARNING - [JSC_TYPE_MISMATCH] Spread operator only applies to Iterable types
found   : (Array<string>|null)
required: Iterable
      conditions: [...context.conditions, 'another-condition'],
                      ^^^^^^^^^^^^^^^^^^

0 error(s), 1 warning(s), 94.1% typed
```

Reply:
please keep the spread operation here.

Reply:
Fixed it, I didn't realize it though null was a possibility (hence null is not iterable).